### PR TITLE
Al tast6to8

### LINKED ISSUE #6: Add `GET` (show) endpoint for a single record in CollegiateSubreddits table
Add `GET` (show) endpoint for a single record in CollegiateSubreddits table

# Acceptance Criteria:

- [x] In `CollegiateSubredditController.java` there is code for an 
      endpoint `GET /api/collegiateSubreddits?id=123` endpoint 
      that returns the JSON of the database record with id 123 if it
      exists, or a 400 and the error message `id 123 not found` if it
      does not.
- [x] The Swagger-UI endpoints for this endpoint is well documented
      so that any member of the team can understand how to use it.
- [x] The endpoint works as expected on localhost.
- [x] The endpoint works as expected when deployed to Heroku.
- [x] There is full test coverage (Jacoco) for the new code in 
      `CollegiateSubredditController.java`
- [x] There is full mutation test coverage (Pitest) for new code in
      `CollegiateSubredditController.java`

# Details

Consult the methods in [`TodosController.java`](https://github.com/ucsb-cs156-w22/demo-spring-react-example-v2/blob/main/src/main/java/edu/ucsb/cs156/example/controllers/TodosController.java) for examples.

Consult the tests in [`TodosControllerTests.java`](https://github.com/ucsb-cs156-w22/demo-spring-react-example-v2/blob/main/src/test/java/edu/ucsb/cs156/example/controllers/TodosControllerTests.java) for examples.

### LINKED ISSUE #7: Add `PUT` (edit) endpoint for a single record in CollegiateSubreddits table
Add `PUT` (edit) endpoint for a single record in CollegiateSubreddits table

Note that it is a good idea to implement `GET` for a single item
first; when using `PUT` manually, it's convenient to be able to copy/paste the JSON representation of the data and then edit it in the swagger interface.

Eventually, it's typically the case that in order to populate an edit
form in the front end, you first do a `GET` on a particular item, 
and use that JSON to populate the fields in a form.  You can send the
JSON representation of the new item to the backend.

# Acceptance Criteria:

- [x] In `CollegiateSubredditController.java` there is code for an 
      endpoint `PUT /api/collegiateSubreddits?id=123` endpoint 
      that accepts JSON for a new set of values for the database
      fields other than `id`, and updates the values of those fields.
- [x] The Swagger-UI endpoints for this endpoint is well documented
      so that any member of the team can understand how to use it.
- [x] The endpoint works as expected on localhost.
- [x] The endpoint works as expected when deployed to Heroku.
- [x] There is full test coverage (Jacoco) for the new code in 
      `CollegiateSubredditController.java`
- [x] There is full mutation test coverage (Pitest) for new code in
      `CollegiateSubredditController.java`

# Details

Consult the methods in [`TodosController.java`](https://github.com/ucsb-cs156-w22/demo-spring-react-example-v2/blob/main/src/main/java/edu/ucsb/cs156/example/controllers/TodosController.java) for examples.

Consult the tests in [`TodosControllerTests.java`](https://github.com/ucsb-cs156-w22/demo-spring-react-example-v2/blob/main/src/test/java/edu/ucsb/cs156/example/controllers/TodosControllerTests.java) for examples.

### LINKED ISSUE #8: Add `DELETE` endpoint for a specific record in CollegiateSubreddits table.
Add `DELETE` endpoint for a specific record in CollegiateSubreddits table.

# Acceptance Criteria:

- [x] In `CollegiateSubredditController.java` there is code for an 
      endpoint `DELETE /api/collegiateSubreddits?id=123` endpoint 
      that deletes the record if it exists, and returns 200 (ok) and 
      the text `record 123 deleted`, or returns 400 (Bad Request) and
      the text `record 123 not found` if it does not.
- [x] The Swagger-UI endpoints for this endpoint is well documented
      so that any member of the team can understand how to use it.
- [x] The endpoint works as expected on localhost.
- [x] The endpoint works as expected when deployed to Heroku.
- [x] There is full test coverage (Jacoco) for the new code in 
      `CollegiateSubredditController.java`
- [x] There is full mutation test coverage (Pitest) for new code in
      `CollegiateSubredditController.java`

# Details

Consult the methods in [`TodosController.java`](https://github.com/ucsb-cs156-w22/demo-spring-react-example-v2/blob/main/src/main/java/edu/ucsb/cs156/example/controllers/TodosController.java) for examples.

Consult the tests in [`TodosControllerTests.java`](https://github.com/ucsb-cs156-w22/demo-spring-react-example-v2/blob/main/src/test/java/edu/ucsb/cs156/example/controllers/TodosControllerTests.java) for examples.

### PR DESCRIPTION
**Pull request by alvin0220**
# Overview

working on the test case issues still need to be work on. There is some issue with the id input.

### CODE REVIEW

File: src/test/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditControllerTests.java [hunk 1]
```diff
@@ -112,4 +112,118 @@ public void api_collegiateSubreddits_post__user_logged_in() throws Exception {
 
     // __________________________________________________________________________________________________________
     
+	// Authorization tests for /api/collegiateSubreddits
+
+    @Test
+    public void api_collegiateSubreddits__logged_out__returns_403() throws Exception {
+        mockMvc.perform(get("/api/collegiateSubreddits"))
```
Comment (AHuangHe) [inline src/test/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditControllerTests.java]: Here, you should be performing the authorization test for a specific action
I'm not sure which one you wanted to test here, but it should be something like:
mockMvc.perform(get("/api/collegiateSubreddits**/all**"))

or, in the case of post, like:
mockMvc.perform(**post**("/api/collegiateSubreddits**/post**"))

File: src/test/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditControllerTests.java [hunk 2]
```diff
@@ -112,4 +112,118 @@ public void api_collegiateSubreddits_post__user_logged_in() throws Exception {
...
+                .andExpect(status().is(403));
+    }
+
+    @WithMockUser(roles = { "USER" })
+    @Test
+    public void api_collegiateSubreddits__logged_in__returns__200() throws Exception {
+        mockMvc.perform(get("/api/collegiateSubreddits"))
```
Comment (AHuangHe) [inline src/test/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditControllerTests.java]: Same case here, it should be more specific. 
Look carefully on the todoscontrollertests file, from lines 42 to 92

File: src/test/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditControllerTests.java [hunk 3]
```diff
@@ -112,4 +112,118 @@ public void api_collegiateSubreddits_post__user_logged_in() throws Exception {
...
+                .andExpect(status().isOk());
+    }
+
+	//delet test 200
+	@WithMockUser(roles = { "USER" })
+    @Test
+    public void api_CollegiateSubreddit_logged_in__delete_collegiateSubreddit() throws Exception {
+        // arrange
+
```
Comment (AHuangHe) [inline src/test/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditControllerTests.java:120]: Here, you forgot to build the collegiate subreddit entity
see lines 301 in the todoscontrollertests file or lines 63-65 from this same file on how to build a collegiatesubreddit entity

File: src/test/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditControllerTests.java [hunk 4]
```diff
@@ -112,4 +112,118 @@ public void api_collegiateSubreddits_post__user_logged_in() throws Exception {
...
+        when(collegiateSubredditRepository.findById(eq(1L))).thenReturn(Optional.empty());
```
Comment (AHuangHe) [inline src/test/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditControllerTests.java]: after building the collegiatesubreddit entity, you should be changing the last bit of this line to:
thenReturn(Optional.**of(subreddit1)**); or however you named the object

File: src/test/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditControllerTests.java [hunk 5]
```diff
@@ -112,4 +112,118 @@ public void api_collegiateSubreddits_post__user_logged_in() throws Exception {
...
+        assertEquals("CollegiateSubreddit with id 1 deleted", responseString);
+    }
+
+	//delete test 403
+	@WithMockUser(roles = { "USER" })
+    @Test
+    public void api_CollegiateSubreddit_logged_in__delete_collegiateSubreddit_that_does_not_exist() throws Exception {
+        // arrange
+
+
```
Comment (AHuangHe) [inline src/test/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditControllerTests.java:153]: I think you have to build another collegiatesubreddit entity here
I think the next line is what makes it seem like the object doesn't actually exist 
See line 323 in todoscontroller

File: src/main/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditController.java [hunk 6]
```diff
@@ -65,4 +81,77 @@ public CollegiateSubreddit postCollegiateSubreddit(
...
+
+        CollegiateSubredditOrError cse = new CollegiateSubredditOrError(id);
+
+        cse = doesCollegiateSubredditOrErrorExist(cse);
+        if (cse.error != null) {
+            return cse.error;
+        }
+      
+
+        incomingCollegiateSubreddit.setId(id);
```
Comment (AHuangHe) [inline src/main/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditController.java:110]: I am not sure, but I think this line might be causing the "Invalid arguments" error
Try commenting it and see if it works

**Review by AHuangHe — CHANGES_REQUESTED** [review]

**Review by AHuangHe — APPROVED** [review]
LGTM

**Comment by Wenjin09** [comment]
good to go

**Pull request merged**